js/content/store.js

```diff
@@ -2589,25 +2590,33 @@ let SearchPageClass = (function(){
 
         // Required for the loading wrapper
         DOMHelper.insertHomeCSS();
+        DOMHelper.insertStyle(".search_pagination_right { display: none; }");
 
-        let result_count;
-        document.querySelector(".search_pagination_right").style.display = "none";
-
-        let match = document.querySelector(".search_pagination_left").textContent.trim().match(/(\d+)(?:\D+(\d+)\D+(\d+))?/);
-        if (match) {
-            result_count = match[2] ? Math.max.apply(Math, match.slice(1, 4)) : match[1];
-            document.querySelector(".search_pagination_left").textContent = Localization.str.results.replace("__num__", result_count);
-        }
+        window.addEventListener("scroll", function() {
+            let node = document.querySelector(".search_pagination_left");
+            if (!node) { return; }
 
-        searchPage = 2;
+            let result_count;
+            let match = node.textContent.trim().match(/(\d+)(?:\D+(\d+)\D+(\d+))?/);
+            if (match) {
+                result_count = match[2] ? Math.max.apply(Math, match.slice(1, 4)) : match[1];
+                node.textContent = Localization.str.results.replace("__num__", result_count);
+            }
 
-        window.addEventListener("scroll", function () {
             // if the pagination element is in the viewport, continue loading
-            if (Viewport.isElementInViewport(document.querySelector(".search_pagination_left"))) {
-                if (result_count > document.querySelectorAll(".search_result_row").length) {
+            if (Viewport.isElementInViewport(node)) {
+                let curRows = document.querySelectorAll(".search_result_row").length;
+
+                // If previous search results got deleted and are no longer relevant (e.g. search term changed), reset search page number
+                if (curRows < rows) {
```

Comment:
Isn't that always `false`?
I don't see where `rows` is changed other than in the line below

Reply:
Initially `rows` is 0, therefore yes this is always `false` since `curRows` cannot be lower than 0.
So, I think the solution here is to initially set `rows` to the current row count (`curRows`).
What do you think?

Reply:
But `!(curRows < curRows)`

Reply:
basically whenever the number of search result rows gets decreased it should assume the search results were no longer relevant and it should reset the page number.

Reply:
I don't really understand how the number of search results should get decreased